### DISCUSSION TITLE
Run Script option to set result properties on top level

### DISCUSSION THREAD

**Discussion by ched-dev** [directus/directus#18981] [Feature Requests]
### Summary

I'd like to be able to set all properties of the result of my Run Script on the top level so I don't have to rewrite the `operationKey` as often in subsequent steps.

### Basic Example

For my Run Script if I have the following operation settings:

<img width="668" alt="Screen Shot 2023-06-22 at 10 56 49 PM" src="https://github.com/directus/directus/assets/73146636/a4a40867-c7b2-4906-a81d-826fae603396">

I want to be able to access the values as `recipient.email` instead of `data.recipient.email`.

### Motivation

I am using a lot of variables in a Send Email operation and writing `$last.recipient.email` or `data.recipient.email` is less readable than `recipient.email`. A top level property would be more succinct and readable.

**Without** top level properties:
```
Hello {{data.recipient.first_name}} {{data.recipient.last_name}},

You received a message from {{data.sender.first_name}} {{data.sender.last_name}}. Please respond to them at {{data.sender.email}} or {{data.recipient.phone}}.

Thanks!
```

**With** top level properties (removing need for `operationKey.`:
```
Hello {{recipient.first_name}} {{recipient.last_name}},

You received a message from {{sender.first_name}} {{sender.last_name}}. Please respond to them at {{sender.email}} or {{recipient.phone}}.

Thanks!
```

### Detailed Design

A boolean field could be added to "Set properties on top level" which defaults disabled.

### Must Have:
- Checkbox to set properties on top level

### Drawbacks

Could pollute top level with lots of properties, or override other top-level properties like `$trigger`, `$last`, or other operation output with matching `operationKey`s.

### Alternatives

None identified

### Adoption Strategy

A new option which defaults to current behavior of setting result on `operationKey`.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
